x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -38,11 +38,47 @@ export interface IngestPipelineParams {
   run_ml_inference: boolean;
 }
 
+export interface FilteringRule {
+  created_at: string;
+  field: string;
+  id: string;
+  order: number;
+  policy: string;
+  rule: string;
+  updated_at: string;
+  value: string;
+}
+
+export interface FilteringValidation {
+  ids: string[];
+  messages: string[];
+}
+
+export interface FilteringRules {
+  advanced_snippet: {
+    created_at: string;
+    updated_at: string;
+    value: object;
```

Comment:
We shouldn't be typing things as 'object', that's too broad and somewhat useless in TypeScript. If it's an actual object of shape unknown, you want to type it as:
```
Record<string, unknown>;
```